geo/src/algorithm/voronoi_diagram.rs

```diff
@@ -0,0 +1,970 @@
+use std::{fmt, result};
+
+use crate::{coord, Coord, GeoFloat, Line, MultiPoint, Point, Polygon, Rect, Triangle};
+
+use crate::triangulate_delaunay::{
+    DelaunayTriangle, DelaunayTriangulationError, DEFAULT_SUPER_TRIANGLE_EXPANSION,
+};
+use crate::{BoundingRect, TriangulateDelaunay};
+
+type Result<T> = result::Result<T, VoronoiDiagramError>;
+
+#[derive(Debug, Clone)]
+pub struct VoronoiComponents<T: GeoFloat> {
+    pub delaunay_triangles: Vec<Triangle<T>>,
+    pub vertices: Vec<Coord<T>>,
+    pub lines: Vec<Line<T>>,
+    pub neighbours: Vec<(Option<usize>, Option<usize>)>,
+}
+
+pub trait VoronoiDiagram<T: GeoFloat>
+where
+    f64: From<T>,
+{
+    fn compute_voronoi_components(
+        &self,
+        clipping_mask: Option<&Polygon<T>>,
+    ) -> Result<VoronoiComponents<T>>;
+    fn compute_voronoi_diagram(&self, clipping_mask: Option<&Polygon<T>>) -> Result<Vec<Polygon>>;
```

Comment:
this isnt implemented anywhere; remove?

Reply:
I've put this as a placeholder to ideally implement before the PR into upstream.  The `geos` version returns the cells as polygons and this would be a good addition.  We don't need it as we just need the vertices hence, for pushing the development back.